File: app/src/main/java/huce/fit/appreadstories/main/MainImpl.java
package huce.fit.appreadstories.main;

import android.content.Context;

import java.util.HashMap;

public class MainImpl implements MainPresenter {

    private static final String TAG = "MainImpl";
    private final HashMap<Integer, Integer> mFragmentOld = new HashMap<>();
    private final int NUMBER = 1;
    int mCount = NUMBER;

    public MainImpl(MainView mainView) {
    }


    @Override
    public void addFragment(int id) {
        mFragmentOld.put(mCount, id);
    }

    @Override
    public void changeFragment(int id) {
        if (mCount == 0) {
            mCount = NUMBER;
        }
        mFragmentOld.put(++mCount, id);
    }

    @Override
    public int backPressed() {
        return --mCount;
    }

    @Override
    public int getId() {
        return mFragmentOld.get(mCount);
    }

    @Override
    public void removeCount() {
        if (mCount != NUMBER) {
            mFragmentOld.remove(mCount);
        }
    }
}

File: app/src/main/java/huce/fit/appreadstories/main/MainImpl.kt
package huce.fit.appreadstories.main

class MainImpl : MainPresenter {

    companion object {
        private const val TAG: String = "MainImpl"
        private const val NUMBER = 1
    }

    private val fragmentOld = HashMap<Int, Int>()
    private var count = NUMBER

    override fun addFragment(id: Int) {
        fragmentOld[count] = id
    }

    override fun changeFragment(id: Int) {
        if (count == 0) {
            count = NUMBER
        }
        fragmentOld[++count] = id
    }

    override fun backPressed(): Int {
        return --count
    }

    override fun getId(): Int {
        return fragmentOld[count]!!
    }

    override fun removeCount() {
        if (count != NUMBER) {
            fragmentOld.remove(count)
        }
    }
}